openmc/mesh.py

```diff
@@ -126,7 +126,76 @@ def from_xml_element(cls, elem):
             raise ValueError(f'Unrecognized mesh type "{mesh_type}" found.')
 
 
-class RegularMesh(MeshBase):
+class StructuredMesh(ABC):
```

Comment:
Any reason not to have this inherit from `MeshBase` and then have the structured mesh types inherit directly from it instead of from both `StructuredMesh` and `MeshBase`?